plugins/animate/helix.hpp

```diff
@@ -0,0 +1,329 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2024 Scott Moreau <oreaus@gmail.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <wayfire/output.hpp>
+#include <wayfire/opengl.hpp>
+#include <wayfire/core.hpp>
+#include <wayfire/view-transform.hpp>
+#include <wayfire/signal-definitions.hpp>
+#include <wayfire/toplevel-view.hpp>
+#include <wayfire/window-manager.hpp>
+#include <wayfire/view-transform.hpp>
+#include <wayfire/txn/transaction-manager.hpp>
+#include <wayfire/render-manager.hpp>
+#include <wayfire/scene-render.hpp>
+#include <wayfire/util/duration.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+#include "animate.hpp"
+
+
+wf::option_wrapper_t<wf::animation_description_t> helix_duration{"animate/helix_duration"};
+wf::option_wrapper_t<int> helix_strip_height{"animate/helix_strip_height"};
+wf::option_wrapper_t<int> helix_rotations{"animate/helix_rotations"};
+
+static const char *helix_vert_source =
+    R"(
+#version 100
+
+attribute mediump vec3 position;
+attribute mediump vec2 uv_in;
+
+uniform mat4 matrix;
+
+varying highp vec2 uv;
+
+void main() {
+    uv = uv_in;
+    gl_Position = matrix * vec4(position, 1.0);
+}
+)";
+
+static const char *helix_frag_source =
+    R"(
+#version 100
+@builtin_ext@
+@builtin@
+
+precision mediump float;
+
+varying highp vec2 uv;
+
+void main()
+{
+    gl_FragColor = get_pixel(uv);
+}
+)";
+
+namespace wf
+{
+namespace helix
+{
+static std::string helix_transformer_name = "animation-helix";
+using namespace wf::scene;
+using namespace wf::animation;
+class helix_animation_t : public duration_t
+{
+  public:
+    using duration_t::duration_t;
+};
+class helix_transformer : public wf::scene::view_2d_transformer_t
```

Comment:
Is there a reason for deriving from `view_2d_transformer_t` instead of `transformer_base_node_t` directly?